lib/ts/controllers/s-popover.ts

```diff
@@ -275,6 +284,162 @@ namespace Stacks {
             });
         }
     }
+
+    function getPopover(element: Element) : [boolean, PopoverController | null, Element | null, Element | null] {
+        const isPopover = getControllerList(element).contains("s-popover");
+        const controller = Stacks.application.getControllerForElementAndIdentifier(element, "s-popover") as PopoverController | null;
+        const referenceSelector = element.getAttribute("data-s-popover-reference-selector");
+        const referenceElement = referenceSelector ? element.querySelector(referenceSelector) : element;
+        const popoverId = referenceElement ? referenceElement.getAttribute("aria-controls") : null;
+        const popover = popoverId ? document.getElementById(popoverId) : null;
+        return [isPopover, controller, referenceElement, popover];
+    }
+
+    /**
+     * Helper to manually show an s-popover element via external JS
+     * @param element the element the `data-controller="s-popover"` attribute is on
+     */
+    export function showPopover(element: Element) {
+        const [isPopover, controller, _] = getPopover(element);
+
+        if (controller) {
+            controller.show();
+        } else if (isPopover) {
+            element.setAttribute("data-s-popover-show-on-connect", "true");
+        } else {
+            throw `element does not have data-controller="s-popover"`;
+        }
+    }
+
+    /**
+     * Helper to manually hide an s-popover element via external JS
+     * @param element the element the `data-controller="s-popover"` attribute is on
+     */
+    export function hidePopover(element: Element) {
+        const [isPopover, controller, _, popover] = getPopover(element);
+
+        if (controller) {
```

Comment:
See question above regarding usage / dom status. In this case, Do we care much about removing the `is-visible` class? If the element isn't in the dom, it won't be visible anyways.

Reply:
This one is mostly about good housekeeping.  If we're hiding, we're hiding.  I just know we will have some case in our codebase where we do `showPopover` and `hidePopover` in the same breath, and if we don't set users up for success, we're going to have a bad time.

Reply:
Reviewers note: See above comment by bnickel with the explanation. tl;dr: race conditions

I'm cool leaving these in now that I know the background behind it